File: SDKs/amazonaws.com/rds/2014-10-31/go/pkg/models/shared/optiongroupoptionsetting.go
package shared

// OptionGroupOptionSetting
// Option group option settings are used to display settings available for each option with their default values and other information. These values are used with the DescribeOptionGroupOptions action.
type OptionGroupOptionSetting struct {
	AllowedValues                       *string
	ApplyType                           *string
	DefaultValue                        *string
	IsModifiable                        *bool
	IsRequired                          *bool
	MinimumEngineVersionPerAllowedValue []MinimumEngineVersionPerAllowedValue
	SettingDescription                  *string
	SettingName                         *string
}

File: SDKs/amazonaws.com/rds/2013-02-12/typescript/src/sdk/models/shared/optiongroupoptionsetting.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class OptionGroupOptionSetting extends SpeakeasyBase {
  @SpeakeasyMetadata()
  allowedValues?: string;

  @SpeakeasyMetadata()
  applyType?: string;

  @SpeakeasyMetadata()
  defaultValue?: string;

  @SpeakeasyMetadata()
  isModifiable?: boolean;

  @SpeakeasyMetadata()
  settingDescription?: string;

  @SpeakeasyMetadata()
  settingName?: string;
}
